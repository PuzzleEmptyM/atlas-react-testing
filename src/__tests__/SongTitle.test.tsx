import { render } from '@testing-library/react';
import SongTitle from '../components/SongTitle';

describe('SongTitle Component', () => {
  it('renders with title and artist', () => {
    const { asFragment } = render(
      <SongTitle title="Sample Song" artist="Sample Artist" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with longer title and artist', () => {
    const { asFragment } = render(
      <SongTitle
        title="A Very Long Song Title That Might Span Multiple Lines And Should Not Have Any Business Being This Long"
        artist="An Artist With An Equally Long Name That Is Very Impressive"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with special characters in title and artist', () => {
    const { asFragment } = render(
      <SongTitle title="Song Title #1%^:|<:{}?" artist="Artist @ Name&$!}|{<>?}:" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
