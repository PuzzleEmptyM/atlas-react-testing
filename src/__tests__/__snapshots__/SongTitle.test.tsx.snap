// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle Component > renders with longer title and artist 1`] = `
<DocumentFragment>
  <div
    class="text-center"
  >
    <h2
      class="text-lg font-bold"
    >
      A Very Long Song Title That Might Span Multiple Lines And Should Not Have Any Business Being This Long
    </h2>
    <p
      class="text-sm text-gray-500"
    >
      An Artist With An Equally Long Name That Is Very Impressive
    </p>
  </div>
</DocumentFragment>
`;

exports[`SongTitle Component > renders with special characters in title and artist 1`] = `
<DocumentFragment>
  <div
    class="text-center"
  >
    <h2
      class="text-lg font-bold"
    >
      Song Title #1%^:|&lt;:{}?
    </h2>
    <p
      class="text-sm text-gray-500"
    >
      Artist @ Name&$!}|{&lt;&gt;?}:
    </p>
  </div>
</DocumentFragment>
`;

exports[`SongTitle Component > renders with title and artist 1`] = `
<DocumentFragment>
  <div
    class="text-center"
  >
    <h2
      class="text-lg font-bold"
    >
      Sample Song
    </h2>
    <p
      class="text-sm text-gray-500"
    >
      Sample Artist
    </p>
  </div>
</DocumentFragment>
`;
